python/ql/src/semmle/python/security/strings/External.qll

```diff
@@ -223,19 +223,19 @@ class UrlsplitUrlparseTempSanitizer extends Sanitizer {
             or
             full_use.(AttrNode).getObject() = test.getInput().getAUse()
         |
-            clears_taint(_, full_use, test.getTest(), test.getSense())
+            clears_taint(full_use, test.getTest(), test.getSense())
         )
     }
 
-    private predicate clears_taint(ControlFlowNode final_test, ControlFlowNode tainted, ControlFlowNode test, boolean sense) {
-        test_equality_with_const(final_test, tainted, sense)
+    private predicate clears_taint(ControlFlowNode tainted, ControlFlowNode test, boolean sense) {
+        test_equality_with_const(test, tainted, sense)
         or
-        test_in_const_seq(final_test, tainted, sense)
+        test_in_const_seq(test, tainted, sense)
         or
         test.(UnaryExprNode).getNode().getOp() instanceof Not and
         exists(ControlFlowNode nested_test |
             nested_test = test.(UnaryExprNode).getOperand() and
-            clears_taint(final_test, tainted, nested_test, sense.booleanNot())
+            clears_taint(tainted, nested_test, sense.booleanNot())
         )
     }
```

Comment:
Looking over this code again, I'm so confused about `final_test` and what it's supposed to do. Thanks for fixing this up :+1: